### PR TITLE
fix: when asm login, it changed the logged in user to anonymous

### PR DESCRIPTION
**Pull request by scarai-sap**
See https://jira.tools.sap/browse/CXSPA-3974

- use userAccountFacade to get customerId as UserProfileFacade is not work.
- Call get customer id before get new token(agent token

### CODE REVIEW

File: feature-libs/asm/root/services/csagent-auth.service.spec.ts [hunk 1]
```diff
@@ -4,12 +4,12 @@ import {
   AuthActions,
   AuthService,
   AuthToken,
+  FeatureConfigService,
   OAuthLibWrapperService,
   OCC_USER_ID_ANONYMOUS,
   OCC_USER_ID_CURRENT,
   UserIdService,
 } from '@spartacus/core';
-import { UserProfileFacade } from '@spartacus/user/profile/root';
```
Comment (scarai-sap) [inline feature-libs/asm/root/services/csagent-auth.service.spec.ts:12]: UserProfileFacade doesn't work

File: feature-libs/asm/root/services/csagent-auth.service.ts [hunk 2]
```diff
@@ -48,6 +75,15 @@ export class CsAgentAuthService {
     password: string
   ): Promise<void> {
     let userToken: AuthToken | undefined;
+    // Start emulation for currently logged in user
```
Comment (scarai-sap) [inline feature-libs/asm/root/services/csagent-auth.service.ts:79]: use userAccountFacade to get customerId as UserProfileFacade is not work.

Call get customer id before get new token(agent token

**Review by scarai-sap — COMMENTED** [review]
add description

**Review by i53577 — APPROVED** [review]
LGTM

**Pull request merged**